docs/content/guides/dagster/troubleshooting.mdx

```diff
@@ -0,0 +1,120 @@
+
+
+## Op or Asset Failures
+
+One of the most common issues that users will hit in Dagster is either an Asset or Op failure. 
+
+#### What went wrong?
+
+Dagit can be used to review the [loggers](/concepts/logging/loggers#using-loggers) to review the stacktrace to investigate what has failed. A few common cases:
+- The Asset or Op could have received some bad data. For more information see [Validating Data with Dagster](/guides/dagster/dagster_type_factories)
+- An external service might have failed, retries in Dagster can be helpful to mitigate this. For more information see [How to recover from flaky failures using retries](concepts/ops-jobs-graphs/op-retries#op-retries)
+- If a non-python sub-process was used, the raw compute logs can be helpful (for more information click here)
+- If something is going wrong with another python library, linking the logger to capture some of the events 
+
+Helpful hint:
+[Exception chaining may be helpful](https://docs.python.org/3.9/tutorial/errors.html#exception-chaining)
+
+---
+
+### Observability 
+
+Dagster provides built-in logger as well an extendible logging system that can be customized. 
+
+Dagster has structured event, compute, agent health and pod logs that can be published. These logs can help dig into what is not working.
+
+
+<!-- ![Viewing the cereals asset in Dagit](/images/guides/troubleshooting/agents.png) -->
+
+<Image
+alt="Viewing the agent console in Dagit"
+src="/images/guides/troubleshooting/agents.png"
+width={3574}
+height={1962}
+/>
+
+Some helpful logs that are available when using Dagster locally or a Hybrid deployment:
+
+- Agent status is available in the Dagit UI which displays if there is a heartbeat and direct access to raw logs is available 
+- Code server status and logs are directly available 
+- Step & Run worker status is available via agent checks and through the event logs as well as raw logs via the compute logs 
+
+#### Deployment Troubleshooting
+
+Agent and pod logs can be a useful tool for resolving Dagster deployment issues. 
+
+[For more information on Dagster Cloud Hybrid Deploymements](/dagster-cloud/deployment/agents)
+
+---
+
+
+## Sensing Failures 
+
+#### Setting up sensors 
+
+When building your data pipeline in Dagster, you have the opportunity to set up custom sensors that can trigger a warning based on a specific situation. [Run failure sensors](/concepts/partitions-schedules-sensors/sensors#run-failure-sensor) are a tool that can be used to monitor run failures. Dagster also provides [built-in failure sensors](/concepts/partitions-schedules-sensors/sensors#out-of-box-run-failure-sensors) that can trigger either a slack message or an email to notify if a run has failed.    
+
+#### Monitoring Sensors in Dagit
+
+The [Dagit UI](/concepts/partitions-schedules-sensors/sensors#monitoring-sensors-in-dagit) can be used to check on sensors and ensure that they are up and running. 
+--- 
+## Definitions
```

Comment:
🔧 This section feels incomplete. After the last sentence, it might be helpful to include a code example that demonstrates a correct `Definitions` object vs an incorrect object.

🔧 Additionally, try to use the `PyObject` component when referring to Dagster objects. Ex:

```
<PyObject object="Definitions" />

# which will result in:
# [Definitions](/_apidocs/definitions)
```

I've got [a guide in the works](https://github.com/dagster-io/dagster/blob/eab3e939e912d719428887a5e7d8e234256b1cf6/docs/README.md#pyobject) that details more about how to use this component.